### PR TITLE
lightgbm 2.2.3 (new formula)

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Please don't specify the amount of jobs in the makefile. It might cause issues with building.

**Comment by ankane** [comment]
My bad, updated

File: Formula/lightgbm.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      args << "-DOpenMP_C_FLAGS=\"-Xpreprocessor -fopenmp -I#{libomp.opt_include}\""
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=\"-Xpreprocessor -fopenmp -I#{libomp.opt_include}\""
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
+
+      system "cmake", *args, ".."
+      system "make"
+    end
+    lib.install "lib_lightgbm.so"
```
Comment (fxcoudert) [inline Formula/lightgbm.rb]: Shared libraries on macOS should be `.dylib`, not `.so`

File: Formula/lightgbm.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    bin.install "lightgbm"
```
Comment (fxcoudert) [inline Formula/lightgbm.rb]: Is there no `make install` target?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/lightgbm.rb (hunk 2 above)
Comment (ankane) [reply Formula/lightgbm.rb]: Hey @fxcoudert, thanks for the feedback. There was a `make install` target, so switched to that.

File: Formula/lightgbm.rb (hunk 1 above)
Comment (ankane) [reply Formula/lightgbm.rb]: Fixed this as well with an additional flag.

**Comment by fxcoudert** [comment]
This is great! Merging.

**Comment by ankane** [comment]
Thanks @fxcoudert 🎉

**Pull request closed without merging**